src/Interfaces.jl

```diff
@@ -94,22 +94,57 @@ function methodparts(IT, x::Expr)
     return methodname, args
 end
 
-requiredmethod(IT, nm, args, shouldthrow) = :(Interfaces.implemented($nm, $args, mods) || ($shouldthrow && Interfaces.missingmethod($IT, $nm, $args, mods)))
+"""
+Construct the method signature from a function name and argument types.
+
+    methodsig(:foo, Tuple{Int64, Float64, Any}) == :(foo(::Int64, ::Float64, ::Any))
+"""
+methodsig(f, args) = Expr(:call, f, unconvertargs(args)...)
+
+function requiredmethod(IT, nm, args, shouldthrow)
+    :(Interfaces.implemented($nm, $args, mods) || ($shouldthrow && Interfaces.missingmethod(debuglvl, $IT, $nm, $args, mods)))
+end
 
-function requiredreturn(T, nm, args, shouldthrow, RT_sym, __RT__)
+function requiredreturn(IT, nm, args, shouldthrow, RT_sym, __RT__)
     return quote
-        check = $(requiredmethod(T, nm, args, shouldthrow))
+        $(requiredmethod(IT, nm, args, shouldthrow))
         $RT_sym = Interfaces.returntype($nm, $args)
         # @show $RT_sym, $nm, $args, Interfaces.isinterfacetype($__RT__)
-        check |= Interfaces.isinterfacetype($__RT__) ?  Interfaces.implements($RT_sym, $__RT__) : $RT_sym <: $__RT__
-        check || ($shouldthrow && Interfaces.invalidreturntype($nm, $args, $RT_sym, $__RT__))
+        check = Interfaces.isinterfacetype($__RT__) ?  Interfaces.implements($RT_sym, $__RT__) : $RT_sym <: $__RT__
+        check || ($shouldthrow && Interfaces.invalidreturntype(debuglvl, $IT, $nm, $args, $RT_sym, $__RT__))
+        check
     end
 end
 
-@noinline missingmethod(IT, f, args, mods) = throw(InterfaceImplementationError("missing `$IT` interface method definition: `$(Expr(:call, f, unconvertargs(args)...))`, in module(s): `$mods`"))
-@noinline invalidreturntype(IT, f, args, RT1, RT2) = throw(InterfaceImplementationError("invalid return type for `$IT` interface method definition: `$(Expr(:call, f, unconvertargs(args)...))`; inferred $RT1, required $RT2"))
-@noinline subtypingrequired(IT, T) = throw(InterfaceImplementationError("interface `$IT` requires implementing types to subtype, like: `struct $T <: $IT`"))
-@noinline atleastonerequired(IT, expr) = throw(InterfaceImplementationError("for `$IT` interface, one of the following method definitions is required: `$expr`"))
+@noinline function missingmethod(debuglvl, IT, f, args, mods)
+    msg = "missing `$IT` interface method definition: `$(Interfaces.methodsig(f, args))`, in module(s): `$mods`"
+    debuglvl == :error ? print_error(msg) : debuglvl == :warn ? print_warn(msg) : throw(InterfaceImplementationError(msg))
```

Comment:
i'd like to find a better option than copy-pasting this line into every error function... but it's not too bad if this is the best we can do for now